qcodes/instrument_drivers/stanford_research/SR86x.py

```diff
@@ -156,17 +158,53 @@ def snapshot_base(self, update: bool = False,
         snapshot = super().snapshot_base(update, params_to_skip_update)
         return snapshot
 
-    @staticmethod
-    def _set_capture_len_parser(value: int) -> int:
+    def _set_capture_len_parser(self, capture_len_in_kb: int) -> int:
+        """
+        According to the manual, the capture length is set in kB. This method checks for the range of the input value,
+        and throws an error if the range is not satisfied. Additionally, if the input value is odd, it is made even
+        (and a warning is issued) since the capture length can only be set in even numbers.
 
-        if value % 2:
-            log.warning("the capture length needs to be even. Setting to {}".format(value + 1))
-            value += 1
+        Args:
+            capture_len_in_kb (int): The desired capture length in kB.
 
-        if not 1 <= value <= 4096:
-            raise ValueError("the capture length should be between 1 and 4096")
+        Returns:
+            capture_len_in_kb (int)
+        """
+        return self._parse_capture_length(capture_len_in_kb, issue_warning=True)
+
+    def _parse_capture_length(self, capture_length_in_kb: int, issue_warning: bool=False) -> int:
```

Comment:
I think this function has a wrong name; it does not really parse anything. Additionally, I think maybe William is right when he says that the function should return exactly what the user asked (see his comment below). In this case, I think the function should be "validate_capture_length" and should not return anything

Reply:
@WilliamHPNielsen Yes, this has all been tested on real hardware